### DISCUSSION TITLE
Make static extension methods anywhere

### DISCUSSION THREAD

**Discussion by chrisseanhayes** [dotnet/csharplang#7515] [Language Ideas]
not sure about the underlying reason why static extension methods need to be in a static class by themselves but i thought it would be nice to be able to add static extension methods in any class. the 'this' keyword for first argument behaves as you would expect.

**Comment by KennethHoff** [comment]
File scoped static classes are a nice workaround for this, but the general solution that the c# team is going for is [extensions](https://github.com/dotnet/csharplang/blob/main/proposals/extensions.md)

**Comment by theunrepentantgeek** [comment]
> not sure about the underlying reason why static extension methods need to be in a static class by themselve

Limiting the search scope to only static classes dramatically reduces the number of candidates that need to be considered when looking for a potential extension method match. 

Given that such a lookup needs to be done for (nearly) every method call during compilation, there's significant potential to negatively impact the performance of the compiler. 

(I have a vague recollection that performance was one of the reasons given when Extension Methods were first introduced, but I can't find anything to corroborate that.)
